app/src/main/java/com/nutomic/syncthingandroid/fragments/DrawerFragment.java

```diff
@@ -206,10 +206,15 @@ private void onReceiveConnections(Connections connections) {
      */
 
     private void showQrCode() {
+        RestApi restApi = mActivity.getApi();
+        if (restApi == null) {
+            Toast.makeText(mActivity, R.string.syncthing_terminated, Toast.LENGTH_SHORT).show();
+            return;
+        }
+        String apiKey = restApi.getGui().apiKey;
+        String deviceId = restApi.getLocalDevice().deviceID;
```

Comment:
Wouldn't these throw exceptions if syncthing was shutdown after checking rest api but before calling getLocalDevice etc?

Reply:
Maybe that can happen, but the shutdown then has to occur simultaneously from another thread which we currently don't have.

Reply:
Any suggestions how to improve this or can we merge it as a first-aid?

Reply:
All of this startup, shutdown, stuff and element access should be behind a lock, there is a ton of NPE's happening because we just happen to access these global things without any locks at random times.

Reply:
Perhaps wrap this whole thing in a try catch, and say an error has occoured if you get an exception.